src/sentry_plugins/github/endpoints/webhook.py

```diff
@@ -99,7 +99,10 @@ def __call__(self, organization, event):
                                 author_email = user.email
                                 gh_username_cache[gh_username] = author_email
                                 if commit_author is not None:
-                                    commit_author.update(email=author_email)
+                                    commit_author.update(
+                                        email=author_email,
+                                        external_id=gh_username,
```

Comment:
Hmm, I wonder if we should specify the provider in the `external_id`. Something like: `'github:%s' % gh_username`. I feel like there'd be a problem when we have multiple providers with the same username across them.